### PR TITLE
Added design doc file for template matching of issues

### LINKED ISSUE #48: Design for Template recognition
## Description
As a user,
I need to create a design for issue template recognition,
so that I can ensure proper formatting for issue creation is followed.

### Update [Required]
- [ ] Create an appropriate template for issue creation formatting checking.
- [ ] Take in more suggestions from the community to finalize.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
Will be finalized after the design is solidified post community discussions.

### PR DESCRIPTION
**Pull request by sammy1997**
# Description
Added design doc file for checking the issue bodies and matching them to see if they follow the standards.

Fixes #48 

# Type of Change:
- Documentation
- This change requires a documentation update (software upgrade on readme file)

# How Has This Been Tested?
view_of_design_readme

# Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by sammy1997** [comment]
@ramitsawhney27 I have added the doc file, as was reviewed.

File: docs/template_matching_design.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Admin: Prachi Manchanda  
+
+###Design for Template matching
+**June 02, 2018**  
+
+Following are the links to a sample issue which follows the template
+and the body of the issue as sent to the
+server by the web-hook:  
+
+[Issue](https://github.com/systers/sysbot-test/issues/48)
```
Comment (sara-02) [inline docs/template_matching_design.md]: I am getting a 404 error for this link.

File: docs/template_matching_design.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+From this final curated list of tokens,
+we need to iterate through the tokens and search for the above given words in the tokens in each run.  
+
+If all are present, along with some text in the next token then we can say that this is a valid template.  
+
+I would like to discuss the course of action that is to be taken if invalid issues are found.  
+
+Should they be ***closed by the bot***?  
+Or should I add a ***label of “Template Mismatch” to them***,
```
Comment (sara-02) [inline docs/template_matching_design.md]: I think the first one is better, we can create a comment saying something like `Your Issue does not follow the template, please update & reopen`

File: docs/template_matching_design.md (hunk 1 above)
Comment (sara-02) [reply docs/template_matching_design.md]: Done thanks.

File: docs/template_matching_design.md (hunk 2 above)
Comment (sara-02) [reply docs/template_matching_design.md]: @sammy1997 As per the discussion we will be following the label method. And one thing we should keep in mind is that if an issue is labelled `template mismatch` then it should neither be `approved` or `claimed.`

Comment (sara-02) [reply docs/template_matching_design.md]: Add that remark in the doc and then we are good to merge this.

**Comment by sammy1997** [comment]
@sara-02 I have updated the doc as per our discussion in the weekly meet

**Pull request merged**